src/lib.rs

```diff
@@ -354,16 +354,21 @@ pub trait PolynomialCommitment<F: PrimeField, P: Polynomial<F>, S: Cryptographic
         Self::Commitment: 'a,
     {
         let BatchLCProof { proof, evals } = proof;
-
         let lc_s = BTreeMap::from_iter(linear_combinations.into_iter().map(|lc| (lc.label(), lc)));
 
         let poly_query_set = lc_query_set_to_poly_query_set(lc_s.values().copied(), eqn_query_set);
+        let sorted_by_poly_and_query_label: BTreeSet<_> = poly_query_set
+            .iter()
+            .map(|(poly_label, v)| ((poly_label.clone(), v.1.clone()), v.0.clone()))
+            .collect();
+
         let poly_evals = Evaluations::from_iter(
-            poly_query_set
+            sorted_by_poly_and_query_label
                 .iter()
-                .map(|(_, point)| point)
-                .cloned()
-                .zip(evals.clone().unwrap()),
+                .zip(evals.clone().unwrap())
+                .map(|(((poly_label, point), _query_label), eval)| {
+                    ((poly_label.clone(), point.clone()), eval)
+                }),
```

Comment:
Let's change the above `iter` also into an `into_iter()`.

```suggestion
                .map(|(((poly_label, point), _query_label), eval)| {
                    ((poly_label, point), eval)
                }),
```